src/v/coproc/reconciliation_backend.cc

```diff
@@ -195,7 +223,72 @@ reconciliation_backend::process_update(model::ntp, update_t delta) {
     __builtin_unreachable();
 }
 
-ss::future<> reconciliation_backend::delete_non_replicable_partition(
+ss::future<std::error_code> reconciliation_backend::process_shutdown(
+  model::ntp src,
+  model::ntp ntp,
+  model::revision_id rev,
+  std::vector<model::broker_shard> new_replicas) {
+    vlog(coproclog.info, "Processing shutdown of: {}", ntp);
+    auto ids = co_await _pacemaker.local().shutdown_partition(src);
+    vlog(coproclog.debug, "Ids {} shutdown");
+    co_await _shard_table.invoke_on_all(
+      [ntp, rev](cluster::shard_table& st) { st.erase(ntp, rev); });
+    auto copro_partition = _coproc_pm.local().get(ntp);
+    if (copro_partition && copro_partition->get_revision_id() < rev) {
+        co_await _coproc_pm.local().remove(ntp);
+    }
+    co_return make_error_code(errc::success);
+}
+
+ss::future<std::error_code> reconciliation_backend::process_restart(
```

Comment:
Would it be possible to add an overview of partition movement logic in the commit message ?